### PR TITLE
tests: Use projects to create and delete test namespace on OpenShift clusters

### PR DESCRIPTION
**Pull request by spadgett**
@dtaylor113 can you try this and let me know if it lets you run against the shared cluster? (You'll need to set the `openshift` param.)

See https://jira.coreos.com/browse/CONSOLE-830

### CODE REVIEW

File: frontend/integration-tests/protractor.conf.ts [hunk 1]
```diff
@@ -78,8 +78,11 @@ export const config: Config = {
     });
     console.log('END BROWSER LOGS');
 
+    // Use projects if OpenShift so non-admin users can run tests. We need the fully-qualified name
+    // since we're using kubectl instead of oc.
+    const resource = browser.params.openshift === 'true' ? 'projects.project.openshift.io' : 'namespaces';
     await browser.close();
-    execSync(`kubectl delete --cascade ns ${testName}`);
```
Comment (spadgett) [inline frontend/integration-tests/protractor.conf.ts]: `--cascade` is unnecessary when deleting namespaces

**Comment by dtaylor113** [comment]
Hi @spadgett, I 'think' this works.  I succesfully ran 
```
yarn run test-suite --suite annotation --params.openshift true --params.servicecatalog true
1) Basic console test
   ✔ logs into console if necessary
   ✔ creates test namespace test-lfknu if necessary

```

However, when I ran the filter suite, it failed with a namespace error:
image

In both cases, I can't see a project or namespace `test-lfknu` or `test-lfknu` on the shared cluster, but I couldn't see them before this PR, so hard to know if they were deleted correctly.

I also tried the `crud` test suite, which failed, but it always failed when I run it locally.

File: frontend/integration-tests/tests/base.scenario.ts [hunk 2]
```diff
@@ -29,15 +29,17 @@ describe('Basic console test', () => {
   });
 
   it(`creates test namespace ${testName} if necessary`, async() => {
```
[the remark this answers is not in the document]
Comment (spadgett) [reply frontend/integration-tests/tests/base.scenario.ts]: Yeah, I get what you're saying. Technically it's a project is also a namespace, so I'm not sure it's really wrong. The con is that we have to pull some of the logic outside of the function / duplicate the check to change the wording.

**Comment by spadgett** [comment]
Hi @dtaylor113. That's because you don't have authority to create daemon sets. That's the second part to the JIRA. I think that should be fixed in a separate PR because it's a larger change.

I think you should be able to run your service catalog suite, though.

**Comment by dtaylor113** [comment]
> I think you should be able to run your service catalog suite, though.

ok, I rebased this PR over my service-catalog sys-test PR, had to comment out any test listing brokers 'cause I don't have privileges to list brokers on shared cluster.  The good news is my service-catalog sys-integration tests passed with this PR, the bad news is that it doesn't help me debug why they fail when run via Jenkins :-(

**Comment by spadgett** [comment]
Opened #536 to use deployments instead of daemon sets for some suites.

**Pull request merged**